Tools/Scripts/webkitpy/layout_tests/controllers/single_test_runner.py

```diff
@@ -175,68 +173,68 @@ def _run_rebaseline(self):
         self._overwrite_baselines(driver_output)
         return TestResult(self._test_input, failures, driver_output.test_time, driver_output.has_stderr(), pid=driver_output.pid)
 
-    _render_tree_dump_pattern = re.compile(r"^layer at \(\d+,\d+\) size \d+x\d+\n")
-
     def _add_missing_baselines(self, test_result, driver_output):
         missingImage = test_result.has_failure_matching_types(test_failures.FailureMissingImage, test_failures.FailureMissingImageHash)
         if test_result.has_failure_matching_types(test_failures.FailureMissingResult):
-            self._save_baseline_data(driver_output.text, '.txt', self._location_for_new_baseline(driver_output.text, '.txt'))
+            self._save_baseline_data(driver_output.text, '.txt')
         if test_result.has_failure_matching_types(test_failures.FailureMissingAudio):
-            self._save_baseline_data(driver_output.audio, '.wav', self._location_for_new_baseline(driver_output.audio, '.wav'))
+            self._save_baseline_data(driver_output.audio, '.wav')
         if missingImage:
-            self._save_baseline_data(driver_output.image, '.png', self._location_for_new_baseline(driver_output.image, '.png'))
-
-    def _location_for_new_baseline(self, data, extension):
-        if self._options.add_platform_exceptions:
-            return self.VERSION_DIR
-        if extension == '.png':
-            return self.PLATFORM_DIR
-        if extension == '.wav':
-            return self.ALONGSIDE_TEST
-        if extension == '.txt' and self._render_tree_dump_pattern.match(data):
-            return self.PLATFORM_DIR
-        return self.ALONGSIDE_TEST
+            self._save_baseline_data(driver_output.image, '.png')
 
     def _overwrite_baselines(self, driver_output):
-        location = self.VERSION_DIR if self._options.add_platform_exceptions else self.UPDATE
-        self._save_baseline_data(driver_output.text, '.txt', location)
-        self._save_baseline_data(driver_output.audio, '.wav', location)
+        self._save_baseline_data(driver_output.text, '.txt', rebaselining=True)
+        self._save_baseline_data(driver_output.audio, '.wav', rebaselining=True)
         if self._should_run_pixel_test:
-            self._save_baseline_data(driver_output.image, '.png', location)
+            self._save_baseline_data(driver_output.image, '.png', rebaselining=True)
 
-    def _save_baseline_data(self, data, extension, location):
+    _render_tree_dump_pattern = re.compile(br"^layer at \(\d+,\d+\) size \d+x\d+\n")
```

Comment:
Might be more clear for this to be at the top of the class, so it's obvious that it's intended to belong to the class instead of an indentation mistake in `_overwrite_baselines`.

Reply:
I've moved this entirely out of the class, given there's not really any reason for it to be within it.